### PR TITLE
[DA-3768] Send slack alert if Intake API payload contains withdrawn PIDs

### PR DESCRIPTION
**Pull request by shahwaiz14**
## Description of changes/additions
- Added cloud tasks that will send slack alert to `rdr-nph-alerts` channel.
- Added function in Participant Dao that will check and return PIDs of withdrawn participant, given a list of PIDs. 

## Tests
✅  unit tests

### CODE REVIEW

File: rdr_service/api/ancillary_study_cloud_tasks_api.py [hunk 1]
```diff
@@ -171,25 +172,21 @@ def post(self):
...
+        log_msg = f"{len(withdrawn_pids)} withdrawn PIDs received in NPH intake payload."
         logging.info(log_msg)
-        json_payload: Dict[str, Any] = self.data
-        create_nph_incident(save_incident=True, slack=True, **json_payload)
+
+        ts = CLOCK.now().strftime('%Y-%m-%d %H:%M:%S')
+        msg = (f"NPH Intake API Payload received on {ts} contains {len(withdrawn_pids)} withdrawn participant IDs:"
+               f" {', '.join(withdrawn_pids)}")
+        create_nph_incident(slack=True, message=msg)
         return {"success": True}
```
Comment (dtharpeuno) [inline rdr_service/api/ancillary_study_cloud_tasks_api.py]: So this is super specific to Participant Ids being withdrawn. Can we make this more general as it was before and either create a new cloud task endpoint just for withdrawn actions?

File: rdr_service/resource/main.py [hunk 2]
```diff
@@ -222,10 +222,11 @@ def _build_resource_app():
...
 
-    # Cloud Task for NPH Incidents
-    _api.add_resource(ancillary_study_cloud_tasks_api.InsertNphIncidentTaskApi,
-                      TASK_PREFIX + "InsertNphIncidentTaskApi",
-                      endpoint="insert_nph_incident_task", methods=["POST"])
+    # Cloud Task for NPH Incidents; currently used to send slack alerts if
+    # withdrawn PIDs sent in NPH Intake Api payload
+    _api.add_resource(ancillary_study_cloud_tasks_api.NphIncidentTaskApi,
+                      TASK_PREFIX + "NphIncidentTaskApi",
+                      endpoint="nph_incident_task", methods=["POST"])
```
Comment (dtharpeuno) [inline rdr_service/resource/main.py]: Same here with the comment about making this more generalized.

File: tests/dao_tests/test_participant_dao.py [hunk 3]
```diff
@@ -506,6 +506,31 @@ def test_error_log_for_incorrect_status_on_participant(self, mock_logging):
...
+        for pid, status in participants_status.items():
+            bid = pid
+            self.dao.insert(
+                Participant(
+                    participantId=pid, biobankId=bid, withdrawalStatus=status
+                )
+            )
+        result = self.dao.get_withdrawn_participant_ids(list(participants_status.keys()))
+
+        self.assertEqual(result, expected_result)
```
Comment (dtharpeuno) [inline tests/dao_tests/test_participant_dao.py]: TBH the more tests the better but moving forward I think we can not add tests for utility methods in the daos unless its super specific. Can we also edit this to lean into the provided generators?

File: rdr_service/api/nph_intake_api.py [hunk 4]
```diff
@@ -227,19 +229,30 @@ def handle_data_from_payload(self):
         self.event_dao_map: dict = self.build_event_dao_map()
         entry_obj: EntryObjData = self.iterate_entries()
 
+        pids = [ele["nph_participant_id"] for ele in self.participant_response]
```
Comment (dtharpeuno) [inline rdr_service/api/nph_intake_api.py]: :nit Can we change this to a `.get`.

**Review by dtharpeuno — CHANGES_REQUESTED** [review]

**Review by dtharpeuno — APPROVED** [review]

**Review by j-kanuch — APPROVED** [review]

**Pull request merged**